### PR TITLE
feat: allow array block contain only length

### PR DESCRIPTION
**Pull request by meandnano**
[#264](https://github.com/hamba/avro/issues/264)

### CODE REVIEW

**Review by nrwiersma — CHANGES_REQUESTED** [review]

File: config.go [hunk 1]
```diff
@@ -36,6 +36,13 @@ type Config struct {
 	// Disable caching layer for encoders and decoders, forcing them to get rebuilt on every
 	// call to Marshal() and Unmarshal()
 	DisableCaching bool
+
+	// DisableArrayBytesLength disables the encoding of the length of the array in bytes.
+	// Encoded arrays will be prefixed with only the number of elements in the array in
+	// contrast with default behavior which prefixes the array with the number of elements
+	// and the number of bytes in the array. Both approaches are valid according to the
+	// Avro specification, however not all decoders support the latter.
+	DisableArraySizeBytes bool
```
[the remark this answers is not in the document]
Comment (meandnano) [reply config.go:45]: I agree. I'm only afraid that it could be confused with the OCF block, but I guess we can expect documentation to clarify this.

**Comment by meandnano** [comment]
One question for another potential PR. Right now there's no way to configure `avro.Writer` inside `ocf.Encoder`. I guess it is possible to add `writerConfig API` field to `ocf.encoderConfig` with corresponding `EncoderFunc` and use this config while constructing `avro.Writer` inside `ocf.NewEncoder`.

**Comment by nrwiersma** [comment]
I assume it is to expose this option, in which case it would be added like `WithBlockLength`.

**Review by nrwiersma — APPROVED** [review]
LGTM 🎉

**Comment by nrwiersma** [comment]
Sounds good to me

**Pull request merged**